### PR TITLE
Add .Net Core 3.1 as target framework for tests

### LINKED ISSUE #3412: Refactor existing UT project to have multiple target platforms - netcore 2.1 , netcore 3.1

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Fixes #3412

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/AssertIgnoreScope.cs [hunk 1]
```diff
@@ -31,20 +31,24 @@ public sealed class AssertIgnoreScope : IDisposable
     {
         public AssertIgnoreScope()
         {
+#if NETFRAMEWORK // Not needed on .Net Core
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/AssertIgnoreScope.cs:34]: I would rephrase this to "Is a NET Framework specific API"

Not needed on .NET core is a big vague (why is it not needed?)

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs [hunk 2]
```diff
@@ -54,10 +55,13 @@ public void SonarAnalysis_WhenShouldAnalysisBeDisabledReturnsTrue_NoIssueReporte
...
             {
-                foreach (var testCase in this.TestCases)
+                foreach (var testCase in TestCases)
                 {
                     // ToDo: We should find a way to ack the fact the action was not run
-                    Verifier.VerifyNoIssueReported(testCase.Path, testCase.Analyzer, ParseOptionsHelper.FromCSharp8);
+                    Verifier.VerifyNoIssueReported(testCase.Path,
+                                                   testCase.Analyzer,
+                                                   ParseOptionsHelper.FromCSharp8,
+                                                   additionalReferences: MetadataReferenceFacade.GetSystemComponentModelPrimitives());
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs:64]: is this metadata necessary for a specific framework or for all?
maybe add a comment?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs [hunk 3]
```diff
@@ -69,24 +73,27 @@ public void SonarAnalysis_WhenShouldAnalysisBeDisabledReturnsTrue_NoIssueReporte
         [TestMethod]
         public void SonarAnalysis_ByDefault_ExecuteRule()
         {
-            foreach (var testCase in this.TestCases)
+            foreach (var testCase in TestCases)
             {
                 // FIX ME: We test that a rule is enabled only by checking the issues are reported
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs:78]: nitpick, clean as you code: replace FIX ME with ToDo?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/CoreMetadataReference.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#if !NETFRAMEWORK
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/CoreMetadataReference.cs:21]: do you plan to have common libraries between the CoreMetadataRefrence and FrameworkMetadataReference in a base class? currently, I see duplication e.g. `Mscorlib`

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+                CoreMetadataReference.MicrosoftVisualBasicCore
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetMicrosoftWin32Registry() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
+#else
+            new[] {CoreMetadataReference.MicrosoftWin32Registry};
+#endif
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs:52]: nitpick add empty line

hm I have on my todo list to look into stylecop

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+        internal static IEnumerable<MetadataReference> GetMicrosoftWin32Primitives() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
+#else
+            new[] {CoreMetadataReference.MicrosoftWin32Primitives};
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemCollections() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs:62]: please add comment "is in mscorlib.dll"

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+                CoreMetadataReference.SystemSecurityCryptographyX509Certificates,
+                CoreMetadataReference.SystemSecurityCryptographyCsp,
+                CoreMetadataReference.SystemSecurityCryptographyCng,
+                CoreMetadataReference.SystemSecurityCryptographyPrimitives
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemSecurityPermissions() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs:148]: please add comment "it's in mscorlib"

well... it's in many places, maybe comment at the top?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DangerousGetHandleShouldNotBeCalledTest.cs [hunk 8]
```diff
@@ -18,7 +18,11 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Collections.Generic;
+using System.Linq;
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DangerousGetHandleShouldNotBeCalledTest.cs:22]: hm is Linq used here?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotInstantiateSharedClassesTest.cs [hunk 9]
```diff
@@ -19,7 +19,10 @@
  */
 
 extern alias csharp;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotInstantiateSharedClassesTest.cs:23]: is Linq used?

Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotInstantiateSharedClassesTest.cs:23]: in general, maybe use some plugin to cleanup unused usings

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/InsecureHashAlgorithmTest.cs [hunk 10]
```diff
@@ -18,6 +18,8 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+#if NETFRAMEWORK // SHA1Managed is sealed in .Net Core and HMACRIPEMD160 is not available
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/InsecureHashAlgorithmTest.cs:21]: maybe we need to explore netcore apis? you could leave an FYI note on the appsec channel

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/NonFlagsEnumInBitwiseOperationTest.cs [hunk 11]
```diff
@@ -30,21 +31,17 @@ public class NonFlagsEnumInBitwiseOperationTest
...
         [TestCategory("CodeFix")]
-        public void NonFlagsEnumInBitwiseOperation_CodeFix()
-        {
-            Verifier.VerifyCodeFix(
-                @"TestCases\NonFlagsEnumInBitwiseOperation.cs",
-                @"TestCases\NonFlagsEnumInBitwiseOperation.Fixed.cs",
-                new NonFlagsEnumInBitwiseOperation(),
-                new NonFlagsEnumInBitwiseOperationCodeFixProvider());
-        }
+        public void NonFlagsEnumInBitwiseOperation_CodeFix() =>
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/NonFlagsEnumInBitwiseOperationTest.cs:41]: hm, how come this doesn't need the `GetSystemComponentModelPrimitives` references?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs [hunk 12]
```diff
@@ -31,40 +31,34 @@ public class RedundantArgumentTest
...
             Verifier.VerifyAnalyzer(@"TestCases\RedundantArgument.cs",
-                new RedundantArgument(),
-                options: ParseOptionsHelper.FromCSharp8,
-                additionalReferences: NuGetMetadataReference.NETStandardV2_1_0);
-        }
+                                    new RedundantArgument(),
+                                    ParseOptionsHelper.FromCSharp8,
+                                    additionalReferences: NuGetMetadataReference.NETStandardV2_1_0);
 
+#if NETFRAMEWORK // ToDo: Fix this when running on .Net Core
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs:40]: what's failing here?

it seems inconsistent that 
- sometimes, you add a issue link
- sometimes, the comment is that an API doesn't work
- here, the comment says to fix

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UsingCookiesTest.cs [hunk 13]
```diff
@@ -35,68 +35,58 @@ namespace SonarAnalyzer.UnitTest.Rules
...
+                additionalReferences: GetNetCoreAdditionalReferences(Constants.DotNetCore220Version));
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void UsingCookies_VB_NetCore() =>
+            Verifier.VerifyAnalyzer(@"TestCases\UsingCookies_NetCore.vb",
+                new VisualBasic.UsingCookies(AnalyzerConfiguration.AlwaysEnabled),
+                additionalReferences: GetNetCoreAdditionalReferences(Constants.DotNetCore220Version));
+
+        private static IEnumerable<MetadataReference> GetNetCoreAdditionalReferences(string packageVersion) =>
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UsingCookiesTest.cs:85]: nitpick: to keep the same name style it should be GetAdditionalReferencesForNetCore

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BeginInvokePairedWithEndInvoke.cs [hunk 14]
```diff
@@ -268,9 +267,15 @@ IAsyncResult BeginInvokeHiddenInALocalFunction()
                     return caller.BeginInvoke("method", null, null); // Noncompliant
                 }
             }
-
         }
-
     }
 
+    internal class AsyncResult : IAsyncResult
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BeginInvokePairedWithEndInvoke.cs:273]: why is this necessary?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs [hunk 15]
```diff
@@ -306,7 +306,7 @@ public static bool ExtensionMethodsCalledForAllDiagnostics(IEnumerable<Diagnosti
                 // why we require each diagnostic to be reported through the extension methods at least once.
                 return analyzers
                     .SelectMany(analyzer => analyzer.SupportedDiagnostics)
-                    .Select(d => counters.GetValueOrDefault(d.Id))
+                    .Select(d => DictionaryExtensions.GetValueOrDefault(counters, d.Id))
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs:309]: why is this change necessary?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs [hunk 16]
```diff
@@ -67,3 +70,18 @@ public UnexpectedDiagnosticException(Location location, string message) : base(m
             Path.GetFullPath(Path.Combine(@"..\..\..\TestCases", path));
     }
 }
+
+#else // Resources cannot be accessed in the same way on .Net Core
+
+namespace SonarAnalyzer.UnitTest.TestFramework
+{
+    public class UnexpectedDiagnosticException : Exception
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs:78]: maybe document somewhere that unit test debugging differs on netcore vs netframework

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs [hunk 17]
```diff
@@ -150,7 +150,7 @@ IEnumerable<TMessage> ReadProtobuf(string path)
         }
 
         public static void VerifyNoIssueReportedInTest(string path, SonarDiagnosticAnalyzer diagnosticAnalyzer,
-            IEnumerable<MetadataReference> additionalReferences = null)
+            IEnumerable<MetadataReference> additionalReferences = null, CompilationErrorBehavior checkMode = CompilationErrorBehavior.Default)
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs:153]: why is this change necessary?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs:64]: I think comments will make the code harder to maintain since we have to maintain both the code and the comments. Also, the answer can be easily found by checking the implementation.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/CoreMetadataReference.cs (hunk 4 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/CoreMetadataReference.cs:21]: I think it would be too much of a change to do this since static classes cannot derive from other classes than Object

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs (hunk 6 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs:62]: I'm not sure these comments help (there are 9 places where we return empty). Isn't it obvious that the types are already available?

Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs:62]: yes, you're actually right

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/CoreMetadataReference.cs (hunk 4 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/CoreMetadataReference.cs:21]: ah indeed

well, we'll see if we need to change this when writing more rules.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs (hunk 7 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs:148]: nevermid this

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotInstantiateSharedClassesTest.cs (hunk 9 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotInstantiateSharedClassesTest.cs:23]: We have to activate **S1128** also for unit tests.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/InsecureHashAlgorithmTest.cs (hunk 10 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/InsecureHashAlgorithmTest.cs:21]: @eric-therond-sonarsource: what do you think about this? should we take any actions?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/NonFlagsEnumInBitwiseOperationTest.cs (hunk 11 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/NonFlagsEnumInBitwiseOperationTest.cs:41]: I'm not sure. The first test is running the rule and fails because it cannot resolve `DesignerSerializationVisibility`. The second one is testing the code fix provider and the `Verifier.VerifyCodeFix` will not check for build errors as `Verifier.VerifyAnalyzer` does.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs (hunk 12 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs:40]: Yes, the answer is simple, it's something I wanted to address and forgot about it.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BeginInvokePairedWithEndInvoke.cs (hunk 14 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BeginInvokePairedWithEndInvoke.cs:273]: In order to be able to keep the tests compatible with both .Net Framework and .Net Core. `System.Runtime.Remoting.Messaging` was used just for adding an implementation for **IAsyncResult** but the implementation is not available on .Net Core so I made a dummy one and removed the namespace.

Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BeginInvokePairedWithEndInvoke.cs:273]: Sounds good, please add a comment

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs (hunk 12 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs:40]: In the end I've created a new issue: https://github.com/SonarSource/sonar-dotnet/issues/3430.
It seems that on .Net Core 3.1 the generated code is incorrect and will not compile.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs (hunk 15 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs:309]: .Net Core provides `System.Collections.Generic.CollectionExtensions.GetValueOrDefault<TKey, TValue>(System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>, TKey)'` which conflicts with `'SonarAnalyzer.Helpers.DictionaryExtensions.GetValueOrDefault<TKey, TValue>(System.Collections.Generic.IDictionary<TKey, TValue>, TKey)'`.

Since in .Net Framework the method from `System.Collections.Generic` is not available, we call `DictionaryExtensions.GetValueOrDefault` as a method instead of extension. This avoids the conflict and usage of conditional code.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs (hunk 16 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs:78]: I don't understand what we gain from this.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs (hunk 17 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs:153]: Not needed. I've removed it.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataReference.cs [hunk 18]
```diff
@@ -77,11 +77,11 @@ internal static class NuGetMetadataReference
         public static IEnumerable<MetadataReference> MicrosoftAspNetCoreHostingAbstractions(string packageVersion) =>
             NugetMetadataFactory.Create("Microsoft.AspNetCore.Hosting.Abstractions", packageVersion);
 
-        public static IEnumerable<MetadataReference> MicrosoftAspNetCoreHttpAbstractions(string packageVersion) =>
+        public static IEnumerable<MetadataReference> MicrosoftAspNetCoreHttpAbstractions(string packageVersion = "2.2.0") =>
             NugetMetadataFactory.Create("Microsoft.AspNetCore.Http.Abstractions", packageVersion);
 
         public static IEnumerable<MetadataReference> MicrosoftAspNetCoreHttpFeatures(string packageVersion) =>
-            NugetMetadataFactory.Create("Microsoft.AspNetCore.Http.Features", packageVersion);
+            NugetMetadataFactory.CreateWithCommandLine("Microsoft.AspNetCore.Http.Features", packageVersion);
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataReference.cs:84]: I had to use `CreateWithCommandLine` since `Create` no longer resolved the reference on my local environment.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs (hunk 16 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs:78]: avoid confusion. For example, if later on, someone will try and run the UTs from a Linux box and would expect a useful debug message, we can tell that person that the Netcore debugging is not as refined as the Net Framework debugging

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs (hunk 12 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/RedundantArgumentTest.cs]: it's a good issue to monitor in the backlog!

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs (hunk 16 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs]: I've realized that we can hardcode the template from English resources for .Net 😄 It could be good enough for now. What do you think?

Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs]: what does "template from English resources for .Net Core" mean?

Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs]: `UnexpectedDiagnosticException` extends the logged information by adding a line which contains the path to the file with errors and the line on which the error occurs. In order to generate this lines it takes two strings from internal resources:
- the word **"at"** by using `mscorlibResources.GetString("Word_At")`
- and the first part of the stack trace, the line which contains the file name and the line number. This is done by `stackTraceFormat = mscorlibResources.GetString("StackTrace_InFileLineNumber")`

The result is a message formatted like: 

> at Test Case in D:\git\sonar-dotnet\sonaranalyzer-dotnet\tests\SonarAnalyzer.UnitTest\TestCases\BooleanCheckInverted.cs:line 22

the same can be achieved by hardcoding the format and the word **"at"**  with the following `$"at Test Case in {path}:line {line}";`.

Since I don't know how we can get the resources on .Net Core, we have to use English words and this means the line will not be translated (to French as example) when running on environments with another cultures.

Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs]: Aaaa yes, I believe it's fine having English messages, it's mainly for our own consumption

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotInstantiateSharedClassesTest.cs (hunk 9 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotInstantiateSharedClassesTest.cs]: fair point, we should look into how to do that (activate rules for unit tests - I believe it's a setting in the RSPEC)
